Tiltfile

```diff
@@ -9,59 +9,71 @@
-dockerfile_js = 'Dockerfile.js.base'
-dockerfile_py = 'Dockerfile.py.base'
-
-# Service: frontend
-fe_img = 'gcr.io/windmill-public-containers/abc123/fe'
-fe_entrypt = '/go/bin/fe --owner ' + username
-
-fast_build(fe_img, dockerfile_go, fe_entrypt).\
-    add(repo.path('fe'), '/go/src/github.com/windmilleng/abc123/fe').\
-    run('go install github.com/windmilleng/abc123/fe')
-k8s_resource(with_username('fe'), new_name='fe', port_forwards=9000)
-
-# Service: letters
-letters_img = 'gcr.io/windmill-public-containers/abc123/letters'
-letters_entrypt = 'node /app/index.js'
-
-fast_build(letters_img, dockerfile_js, letters_entrypt).\
-    add(repo.path('letters/src'), '/app').\
-    add(repo.path('letters/package.json'), '/app/package.json').\
-    add(repo.path('letters/yarn.lock'), '/app/yarn.lock').\
-    run('cd /app && yarn install', trigger=['letters/package.json', 'letters/yarn.lock'])
-k8s_resource(with_username('letters'), new_name='letters')
-
-# Service: numbers
-numbers_img = 'gcr.io/windmill-public-containers/abc123/numbers'
-numbers_entrypt = 'node /app/index.js'
-
-fast_build(numbers_img, dockerfile_py). \
-    add(repo.path('numbers'), '/app'). \
-    run('cd /app && pip install -r requirements.txt', trigger='numbers/requirements.txt')
-k8s_resource(with_username('numbers'), new_name='numbers')
+### NOTE: this Tiltfile / incremental onboarding expereicen currently on work on LOCAL k8s clusters
+### that don't require pushing images -- so, e.g. k8s for Docker for Mac, Minikube + Docker
+
+### Step 0: Hello World ###
+# # Uncomment this to see Tilt do something!
+# print("Welcome to Tilt! 👋")
+
+### Step 1: Kubernetes YAML ###
+# # Start with just your existing Kubernetes yaml -- your services will run (if the
+# # images are available) and you can see their status and logs, but Tilt isn't
+# # building anything for you, so code changes won't be propagated
+# # NOTE: build these images first with `make build` 👀
+# k8s_yaml([
+#     'fe/deployments/fe.yaml',
+#     'letters/deployments/letters.yaml',
+#     'numbers/deployments/numbers.yaml',
+# ])
+
+### Step 2: Port Forwarding ###
+# # Uncomment this line to port-forward your frontend so you can hit it locally
+# k8s_resource('fe', port_forwards='8000')
+
+### Step 3: Tilt Image Build ###
+# # Uncomment these lines to have Tilt build images for your code (and
+# # smartly re-build whenever things change on disk)
+# docker_build('abc123/fe', 'fe')            # == `docker build ./fe -t abc123/fe`
+# docker_build('abc123/letters', 'letters')  # == `docker build ./letters -t abc123/letters`
+# docker_build('abc123/numbers', 'numbers')  # == `docker build ./numbers -t abc123/numbers`
+
+### Step 4: Live Update ###
+# # The frontend builds pretty slowly, doesn't it? That's because every time you change it,
+# # Docker re-builds the container, which means it has to tar up a really big build context
+# # because of that pesky `fe/big_context` directory (16KB). Let's use Live Update instead,
```

Comment:
I'd put a readme in fe/big_context that explains its existence.

Also, I wouldn't expect copying a 16kb file to have that much impact on build speed? I'd expect most of the time would be lost to, like, the apt-get and the go get and the lack of go build cache. Is there much difference without it?

Also might be worth a link to the blog post that explains how live update works, since it's kind of too complicated give a proper explanation here.

Reply:
lol i'm just gonna take out the big context entirely